### PR TITLE
Replace Compass with gulp-sass

### PR DESCRIPTION
**Pull request by scott1702**
Followed on from https://github.com/silverstripe/silverstripe-framework/pull/5027

CMS PR - https://github.com/silverstripe/silverstripe-cms/pull/1393

@dhensby @jonom

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -245,3 +258,73 @@ gulp.task('umd-watch', function () {
     gulp.watch(PATHS.ADMIN_JAVASCRIPT_SRC + '/*.js', ['umd-admin']);
     gulp.watch(PATHS.FRAMEWORK_JAVASCRIPT_SRC + '/*.js', ['umd-framework']);
 });
+
+
+/*
+ * Sprite and scss/css compilation
+ */
+ 
+gulp.task('compile', ['sprites', 'compile-admin:css', 'compile:css', 'compile-dev-install:css'], function () {
```
Comment (chillu) [inline gulpfile.js]: `compile` isn't a great name here since we already have `build`, which should really generate everything (JS, CSS and images). Suggest we rename the task to `css`, with "subtasks" `sprites` and `css`. I don't think there's a need to separate out tasks for `framework/scss`, `framework/admin/scss` and `framework/dev/install/scss`, unless I'm missing something here?

File: gulpfile.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ var gulp = require('gulp'),
     babel = require('gulp-babel'),
```
Comment (chillu) [inline gulpfile.js]: The tasks in here are getting pretty hard to follow, lots going on - can you please start adding a docblock on top of the new tasks describing what they do?

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/05_CMS_Architecture.md [hunk 3]
```diff
@@ -49,6 +52,17 @@ As there's a whole lot of CSS driving the CMS, we have certain best practives ar
 
 See our [system requirements](/getting_started/server_requirements) for a list of supported browsers.
 
+## Sprites
+
+We use sprites to handle various icons and images throughout the CMS. These are automatically generated
+by running `npm run compile` and can be found at `/admin/images/sprites/dist`. To add new
```
Comment (chillu) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/05_CMS_Architecture.md]: Don't forget to update the naming here as well - I suggested `npm run css` instead of `npm run compile`. Might make sense to have a separate `npm run sprites` task.

File: docs/en/04_Changelogs/4.0.0.md [hunk 4]
```diff
@@ -61,6 +61,10 @@ or developing your own website. These improvements are mainly geared at CMS core
    * [react-addons-test-utils](https://www.npmjs.com/package/react-addons-test-utils)
  * Added SilverStripeComponent base class for building React UIs
 
+### Compiling CSS
+
+Compass has been removed and css/sprite compilation is now handled with gulp (see [Markup and Style Conventions](../developer_guides/customising_the_admin_interface/cms_architecture/#markup-and-style-conventions)).
```
Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: Link to both Compass and Gulp projects - not every PHP dev will be on top of the latest buzzwords ;)

**Comment by scott1702** [comment]
Made the changes that you recommended @chillu.

**Comment by tractorcow** [comment]
Please update https://github.com/scott1702/silverstripe-framework/blob/compass-gulp/docs/en/05_Contributing/01_Code.md#npm with the new CSS compilation tasks (and any other npm tasks we've missed out).

**Comment by scott1702** [comment]
Updated @tractorcow

**Comment by tractorcow** [comment]
Restarting tests.
